### ISSUE TITLE
Error when trying to connect to Raindrop

### ISSUE THREAD

**Issue by SebastianAtWork** [kaiiiz/obsidian-raindrop-highlights-plugin#43]
Hi,
I´ve created a brand new vault and only installed obsidian raindrop highlights plugin. On a Windows 11 Machine with Obsidian 1.1.16. I created a fresh Raindrop Application Registration with a new test Token. When trying to manage the synced collections or sync, I get the following Error in the Developer Console:
app.js:1 Obsidian Developer Console
app.js:1 Raindrop Highlights: styles.css not found
plugin:obsidian-raindrop-highlights:12461 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '$id')
    at eval (plugin:obsidian-raindrop-highlights:12461:41)
    at Array.forEach (<anonymous>)
    at RaindropAPI.eval (plugin:obsidian-raindrop-highlights:12457:17)
    at Generator.next (<anonymous>)
    at fulfilled (plugin:obsidian-raindrop-highlights:35:24)
eval @ plugin:obsidian-raindrop-highlights:12461
eval @ plugin:obsidian-raindrop-highlights:12457
fulfilled @ plugin:obsidian-raindrop-highlights:35
Promise.then (async)
step @ plugin:obsidian-raindrop-highlights:47
eval @ plugin:obsidian-raindrop-highlights:48
__async @ plugin:obsidian-raindrop-highlights:32
eval @ plugin:obsidian-raindrop-highlights:12111
(anonymous) @ app.js:1

I´ve attached the zipped vault.
[RaindropVault.zip](https://github.com/kaiiiz/obsidian-raindrop-highlights-plugin/files/11351240/RaindropVault.zip)
(The Token in this Example was made invalid, so has to be re-entered)

**Comment by kaiiiz** [comment]
I've tested your vault setting but can not reproduce the bug you encountered. Can you provide the API responses for the following two endpoints: https://api.raindrop.io/rest/v1/collections and https://api.raindrop.io/rest/v1/collections/childrens?

Example using curl:

```
curl -H "Authorization: Bearer <replace this to your token>" https://api.raindrop.io/rest/v1/collections
```

```
curl -H "Authorization: Bearer <replace this to your token>" https://api.raindrop.io/rest/v1/collections
```

Thanks.

**Comment by SebastianAtWork** [comment]
[CollectionsResponse.txt](https://github.com/kaiiiz/obsidian-raindrop-highlights-plugin/files/11360986/CollectionsResponse.txt)
[CollectionsChildrensResponse.txt](https://github.com/kaiiiz/obsidian-raindrop-highlights-plugin/files/11360987/CollectionsChildrensResponse.txt)
I removed all except to examples from the second since it contains sensible data.

**Comment by SebastianAtWork** [comment]
Im currently debugging and it seems that there is one collection, that has no parent and thus 
 nestedCollectionMap[id] = {
          title: collection["title"],
          parentId: collection["parent"]["$id"]
        };
throws an exception
.
Here is the relevant collection:

`{
    "_id": 31804532,
    "title": "Obsidian",
    "description": "",
    "public": false,
    "view": "list",
    "count": 36,
    "cover": [],
    "expanded": false,
    "user": {
        "$ref": "users",
        "$id": 1207074
    },
    "creatorRef": {
        "_id": 1207074,
        "name": "sebastian-schwarz",
        "email": ""
    },
    "lastAction": "2023-04-12T06:45:03.120Z",
    "created": "2023-02-20T07:52:24.934Z",
    "lastUpdate": "2023-04-12T06:45:03.121Z",
    "sort": 31804532,
    "slug": "obsidian",
    "collaborators": {
        "$ref": "collaborators",
        "$id": "63f4a4ceb0dc6502c1ee0c87"
    },
    "access": {
        "for": 384694,
        "level": 3,
        "root": true,
        "draggable": false
    },
    "author": true
}`

**Comment by kaiiiz** [comment]
Closed due to inactive.

**Comment by SebastianAtWork** [comment]
This is still an issue, even in the newest version

**Comment by kaiiiz** [comment]
Sorry for accidentally closing the issue. It's quite strange that a collection child can have no parent... Can you provide your Raindrop collection structure? I noticed that the user of the "Obsidain" collection is different. Is this collection created by the other user?

**Comment by SebastianAtWork** [comment]
Ahh thanks. That folders owner was my previous work account. I dont have access to that anymore. I exported and imported it. Would you accept a pull request that just uses ?. to check if the parent id is set or not and set it to 0 ? 
api.ts line 106:
```typescript
	const nestedCollectionMap: { [id: number]: NestedRaindropCollection } = {};
		const nestedCollections = await nestedCollectionPromise;
		nestedCollections.items.forEach((collection: any) => {
			const id = collection["_id"];
			nestedCollectionMap[id] = {
				title: collection["title"],
				parentId: collection["parent"]?.["$id"]??0,
			};
		});

		nestedCollections.items.forEach((collection: any) => {
			const id = collection["_id"];
			let parentId = collection["parent"]?.["$id"]??0;
			let title = collection["title"];
```

**Comment by kaiiiz** [comment]
Fixed in v0.0.21.
